src/main/java/org/folio/circulation/domain/representations/changeduedate/BatchChangeDueDateRequest.java

```diff
@@ -0,0 +1,76 @@
+package org.folio.circulation.domain.representations.changeduedate;
+
+import static org.folio.circulation.support.JsonPropertyFetcher.getDateTimeProperty;
+import static org.folio.circulation.support.Result.succeeded;
+import static org.folio.circulation.support.ValidationErrorFailure.failedValidation;
+
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+import java.lang.invoke.MethodHandles;
+import java.util.List;
+import org.folio.circulation.support.Result;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BatchChangeDueDateRequest {
+
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  private static final String LOAN_IDS = "loanIds";
+  private static final String DUE_DATE = "dueDate";
+
+  private final List<String> loanIds;
+  private final DateTime dueDate;
+
+  private BatchChangeDueDateRequest(List<String> loanIds,
+    DateTime dueDate) {
+
+    this.loanIds = loanIds;
+    this.dueDate = dueDate;
+  }
+
+  public static Result<BatchChangeDueDateRequest> from(RoutingContext routingContext) {
+
+    List<String> loanIds;
+    DateTime dueDate;
+    try {
+      JsonObject payload = routingContext.getBodyAsJson();
+      if (payload == null) {
+        return failedValidation("No payload provided",
+          "", null);
+      }
+
+      JsonArray jsonArray = payload.getJsonArray(LOAN_IDS);
+      if (jsonArray == null || jsonArray.isEmpty()) {
+        return failedValidation("No loan ids provided",
+          LOAN_IDS, null);
+      }
+
+      loanIds = jsonArray.getList();
+      dueDate = getDateTimeProperty(payload, DUE_DATE);
+
+      if (dueDate == null) {
+        return failedValidation("No due date provided",
+          DUE_DATE, null);
+      }
+
+    } catch (Exception e) {
```

Comment:
What is the case when the exception might be thrown? If `dueDate` has an invalid format, right?

Reply:
This is just a precausion so that unhandled exceptions are not thrown to the client.

Reply:
To follow on to @bohdan-suprun's point, I am not sure that catching `Exception` here really indicates a 422. If we don't know what is going on, we likely can't expect the client to be able to do much about it. I would think a 500 would be more appropriate here. Also, the validation response indicates that the "message" field was incorrect when there is no "message" key in the schema.

Will allowing an unexpected Exception to be thrown out of this method be caught by something higher up the stack and return a 500 to the client? If so, maybe we catch and log (for better context) then rethrow the exception or just remove the catch.